src/components/PopupWithForm.js

```diff
@@ -31,10 +32,15 @@ export default class PopupWithForm extends Popup {
     super.setEveentListeners();
     this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
+      this._submitButton.textContent = "Save...";
       this._handleFormSubmit(this._getInputValues());
-      this._popupForm.reset();
-      this._validator.toggleButtonState();
-      this.close();
     });
   }
+
+  submissionCompleate() {
+    this._submitButton.textContent = "Save";
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.